daemon/reload.go

```diff
@@ -28,7 +28,17 @@ func (daemon *Daemon) Reload(conf *config.Config) (err error) {
 	attributes := map[string]string{}
 
 	defer func() {
-		jsonString, _ := json.Marshal(daemon.configStore)
+		jsonString, _ := json.Marshal(&struct {
+			*config.Config
+			config.ProxyConfig
+		}{
+			Config: daemon.configStore,
+			ProxyConfig: config.ProxyConfig{
+				HTTPProxy:  config.MaskCredentials(daemon.configStore.HTTPProxy),
+				HTTPSProxy: config.MaskCredentials(daemon.configStore.HTTPSProxy),
+				NoProxy:    config.MaskCredentials(daemon.configStore.NoProxy),
+			},
+		})
```

Comment:
This wasn't introduced here, but now that it's doing more work than just "convert thing to JSON", wouldn't it make sense to move this down inside the `if` statement below so we don't generate it unless we actually use it?

Reply:
Ah, yeah. I guess I put it here in the first place, so that we'd generate the JSON while we still have the lock. I could make this conditional though (only do this if `err ==nil`)